src/libsync/clientsideencryption.cpp

```diff
@@ -1777,4 +1780,179 @@ bool EncryptionHelper::fileDecryption(const QByteArray &key, const QByteArray& i
     return true;
 }
 
+EncryptionHelper::StreamingDecryptor::StreamingDecryptor(const QByteArray &key, const QByteArray &iv, quint64 totalSize) : _totalSize(totalSize)
+{
+    if (_ctx && !key.isEmpty() && !iv.isEmpty() && totalSize > 0) {
+        _isInitialized = true;
+
+        /* Initialize the decryption operation. */
+        if(!EVP_DecryptInit_ex(_ctx, EVP_aes_128_gcm(), nullptr, nullptr, nullptr)) {
+            qCritical(lcCse()) << "Could not init cipher";
+            _isInitialized = false;
+        }
+
+        EVP_CIPHER_CTX_set_padding(_ctx, 0);
+
+        /* Set IV length. */
+        if(!EVP_CIPHER_CTX_ctrl(_ctx, EVP_CTRL_GCM_SET_IVLEN, iv.size(), nullptr)) {
+            qCritical(lcCse()) << "Could not set iv length";
+            _isInitialized = false;
+        }
+
+        /* Initialize key and IV */
+        if(!EVP_DecryptInit_ex(_ctx, nullptr, nullptr, reinterpret_cast<const unsigned char*>(key.constData()), reinterpret_cast<const unsigned char*>(iv.constData()))) {
+            qCritical(lcCse()) << "Could not set key and iv";
+            _isInitialized = false;
+        }
+    }
+}
+
+qint64 EncryptionHelper::StreamingDecryptor::chunkDecryption(const char *input, QIODevice *output, quint64 chunkSize)
+{
+    Q_ASSERT(isInitialized());
+    if (!isInitialized()) {
+        qCritical(lcCse()) << "Decryption failed. Decryptor is not initialized!";
+        return -1;
+    }
+
+    Q_ASSERT(output && output->isOpen() && output->isWritable());
+    if (!output || !output->isOpen() || !output->isWritable()) {
+        qCritical(lcCse()) << "Decryption failed. Incorrect output device!";
+        return -1;
+    }
+
+    Q_ASSERT(input);
+    if (!input) {
+        qCritical(lcCse()) << "Decryption failed. Incorrect input!";
+        return -1;
+    }
+
+    Q_ASSERT(chunkSize > 0);
+    if (chunkSize <= 0) {
+        qCritical(lcCse()) << "Decryption failed. Incorrect chunkSize!";
+        return -1;
+    }
+
+    if (_decryptedSoFar == 0) {
+        qCDebug(lcCse()) << "Decryption started";
+    }
+
+    Q_ASSERT(_decryptedSoFar + chunkSize <= _totalSize);
+    if (_decryptedSoFar + chunkSize > _totalSize) {
+        qCritical(lcCse()) << "Decryption failed. Chunk is out of range!";
+        return -1;
+    }
+
+    Q_ASSERT(_decryptedSoFar + chunkSize < OCC::CommonConstants::e2EeTagSize || _totalSize - OCC::CommonConstants::e2EeTagSize >= _decryptedSoFar + chunkSize - OCC::CommonConstants::e2EeTagSize);
+    if (_decryptedSoFar + chunkSize > OCC::CommonConstants::e2EeTagSize && _totalSize - OCC::CommonConstants::e2EeTagSize < _decryptedSoFar + chunkSize - OCC::CommonConstants::e2EeTagSize) {
+        qCritical(lcCse()) << "Decryption failed. Incorrect chunk!";
+        return -1;
+    }
+
+    const bool isLastChunk = _decryptedSoFar + chunkSize == _totalSize;
```

Comment:
@FlexW Works, but, I don't see any benefit of using it here. `bool `is always bool and is quite short. I wouldn't abuse `auto` if no be nefits are brought by using it.

Reply:
Just find it more consistent and it's clear that it is a bool because of the comparison operator.

Reply:
@FlexW Well, if we want consistency, we must replace all variables with auto. Yet, let's say we are searching for a `bool` variable in all files or one file. Having all types deduced with `auto` wouldn't really look easy to read. But, I may be wrong. Let's see what others think @mgallien @camilasan

Reply:
I would be OK with using `bool` here because the type is not already used and we never really decided to use `auto` almost everywhere
then I would say that @FlexW also has a point in general but large scale conversion should be done with `clang-tidy` before we ask people to almost always use `auto`